x-pack/plugins/fleet/common/services/is_agent_upgradeable.ts

```diff
@@ -29,17 +29,16 @@ export const LATEST_VERSION_NOT_VALID_ERROR = 'latest version is not valid.';
 export const AGENT_ALREADY_ON_LATEST_ERROR = `agent is already running on the latest available version.`;
 export const AGENT_ON_GREATER_VERSION_ERROR = `agent is running on a version greater than the latest available version.`;
 
-export function isAgentUpgradeable(
-  agent: Agent,
-  latestAgentVersion: string,
-  versionToUpgrade?: string
-) {
-  let agentVersion: string;
-  if (typeof agent?.local_metadata?.elastic?.agent?.version === 'string') {
-    agentVersion = agent.local_metadata.elastic.agent.version;
-  } else {
-    return false;
-  }
+export function isAgentUpgradeAvailable(agent: Agent, latestAgentVersion?: string) {
+  return (
+    latestAgentVersion &&
+    isAgentUpgradeable(agent) &&
+    agent?.local_metadata?.elastic?.agent?.version &&
+    isAgentVersionLessThanLatest(agent.local_metadata.elastic.agent.version, latestAgentVersion)
+  );
+}
+
+export function isAgentUpgradeable(agent: Agent) {
```

Comment:
The cypress test caught that the `Upgrade available` filter on agent list broke, it returned an agent on latest agent version.
Decided to do some refactoring and add new methods to separate the logic better:
- isAgentUpgradeable: checks if agent upgrade is possible
- isAgentUpgradeAvailable: checks if there is a newer agent version available
- isAgentUpgradeableToVersion: checks if the agent is upgradeable to the given target version